kernel/src/grant.rs

```diff
@@ -42,28 +39,48 @@ impl<'a, T: 'a + ?Sized> DerefMut for Borrowed<'a, T> {
     }
 }
 
+/// A grant applied to a particular process.
+///
+/// This is created from a `Grant` to setup memory for that grant in a specific
+/// process's grant region. An `AppliedGrant` guarantees that the memory for the
+/// grant has been allocated in the process's memory.
```

Comment:
We probably want to precisely define the difference between a "grant", a "grant region", and what's current an `AppliedGrant`. Perhaps grant region, grant, and grant instance?

Reply:
Yes! This is very confusing.